Odata-docs/webapi-8/fundamentals/query-options.md

```diff
@@ -0,0 +1,568 @@
+                    "Price": 15,
+                    "Quantity": 50
+                },
+                {
+                    "Id": 1005,
+                    "Price": 40,
+                    "Quantity": 15
+                }
+            ]
+        },
+        {
+            "Name": "Customer 2",
+            "Orders": []
+        },
+        {
+            "Name": "Customer 1",
+            "Orders": []
+        }
+    ]
+}
+```
+
+### Limiting Query Options
+We can limit the type of query option that can be used while calling a certain API action.
+
+```csharp
+public class CustomersController : ODataController
+{
+    [EnableQuery(AllowedQueryOptions = AllowedQueryOptions.Filter)]
+    public IActionResult Get()
+    {
+        return Ok(Customers);
+    }
+}
+```
+
+In the controller method above, we only allow `$filter` query option. If we use another query option e.g `$select`, we will get an error.
+
+```http
+GET http://localhost:6285/odata/Customers?$select=Name
+```
+
+```json
+{
+    "error": {
+        "code": "",
+        "message": "The query specified in the URI is not valid. Query option 'Select' is not allowed. To allow it, set the 'AllowedQueryOptions' property on EnableQueryAttribute or QueryValidationSettings.",
+        "details": [],
+        "innererror": {}
+    }
+}
+```
+
+We can combine multiple query options as shown below:
+
+```csharp
+[EnableQuery(AllowedQueryOptions = AllowedQueryOptions.Filter | AllowedQueryOptions.OrderBy)]
+public IQueryable<Customer> Get()
+{
+    return Customers.AsQueryable<Customer>();
+}
+```
+
+## Config options vs AllowedQueryOptions
+When registering OData services, we add query options configurations as shown below:
+
+```csharp
+.AddOData(
+    options => options.Select().OrderBy().Expand().Count().SetMaxTop(null)
+)
+```
+
+These are global configurations. If a query option is not enabled here, it cannot be enabled in the controller.
+
+In the above example, we did not enable the $filter query option. If we configure a controller as shown below, the $filter will not be applied since we had not enabled $filter in the global configuration.
+
+```csharp
+[EnableQuery(AllowedQueryOptions = AllowedQueryOptions.Filter)]
+public IActionResult Get()
+{
+    return Ok(Customers);
+}
+```
+
+## Invoking query options directly
+There are scenarios where we are unable to use `EnableQuery` attribute e.g When inside the controller method, we are getting data from multiple sources and we would want that data to be processed before it's returned to the controller method.
+
+We use the `ODataQueryOptions.ApplyTo` method to apply the required query options.
+
+### SelectExpand query
+```csharp
+public IQueryable<Customer> Get(ODataQueryOptions<Customer> options)
+{
+    IQueryable results = options.ApplyTo(Customers.AsQueryable());
+
+    return results as IQueryable<Customer>;
+}
+```
+In the example above, `options.SelectExpand` property will be set when we make the request below:
```

Comment:
I got a bit confused reading this section. Especially because it mentions`options.SelectExpand` "in the example above" yet it doesn't appear anywhere so for. I think you should first give an example of `options.ApplyTo` on its own and explain that it applies all query options.

Then mention that you can apply query options individually, and then proceed with a section for each query option property like you have done below. Then you don't have to repeat the `options.ApplyTo` for each sub-section.